ops/testing.py

```diff
@@ -1636,18 +1636,22 @@ def get_filesystem_root(self, container: Union[str, Container]) -> pathlib.Path:
         Example usage::
 
             # charm.py
-            import ops
             class ExampleCharm(ops.CharmBase):
                 def __init__(self, *args):
                     super().__init__(*args)
-                    self.hostname = open("/etc/hostname").read()
+                    self.framework.observe(self.on["mycontainer"].pebble_ready,
+                                           self._on_pebble_ready)
+
+                def _on_pebble_ready(self, event: ops.PebbleReadyEvent):
+                    self.hostname = event.workload.pull("/etc/hostname").read()
 
             # test_charm.py
-            from ops.testing import Harness
             harness = Harness(ExampleCharm)
             root = harness.get_filesystem_root("mycontainer")
+            (root / "etc").mkdir()
             (root / "etc" / "hostname").write_text("hostname.example.com")
-            harness.begin()
+            harness.begin_with_initial_hooks()
+            assert harness.charm.hostname == "hostname.example.com"
 
```

Comment:
This is only tangentially related, but while we're looking at this example, do you think it's worth adjusting the example test to call `cleanup()` since one of the key things that does is clean up the temporary directory? Or is that too much noise here and we expect that people will instantiate a Harness object in setUp/a fixture with cleanup there, copying that from somewhere else?

Reply:
Yeah, good call. Updated.